frontend/public/src/containers/pages/register/RegisterConfirmPage.js

```diff
@@ -113,8 +113,12 @@ const mapStateToProps = createStructuredSelector({
   qsParams: qsSelector
 })
 
-const mapPropsToConfig = ({ qsParams }) =>
-  mutateAsync(queries.auth.registerConfirmEmailMutation(qsParams))
+const mapPropsToConfig = ({ qsParams }) => {
+  const { type, ...config } = mutateAsync(
+    queries.auth.registerConfirmEmailMutation(qsParams)
+  )
+  return config
+}
```

Comment:
This was giving an error:
```
Cannot call connectRequest with mapPropsToConfig bound to mapPropsToConfigs
because property type is missing in QueryConfig [1] but exists in
MutateAsyncAction [2] in the return value.
```

Again, Flow did not complain on `main` because redux-query is untyped on main, so anything is allowed.

The error occurs because:
1. Now `redux-query` has Flow types...
2. `redux-query` expects `mapPropsToConfig` to return a QueryConfig, which [their type definition](https://github.com/amplitude/redux-query/blob/562599ee1c482098879ae30e52898862c4639bec/packages/redux-query/src/types.js#L25C13-L25C24) defines as an [exact type](https://flow.org/en/docs/types/objects/#exact-and-inexact-object-types), so **it is not allowed to have extra keys**. But the return value of `mutateAsync` does have an extra key, namely `"type"`.
    - Javascript doesn't care about the extra properties, which is why everything seems to work fine. 
    - We use Flow 0.95, in which exact types are denoted with pipes `type Exact = {| ... |}`. Exact types are now the default in Flow.
    - In contrast, Typescript does not have the concept of exact types. So in Typescript, a declaration like
     ```ts
    type Dog = { name: string } 
    const sayWoof = (dog: Dog) => console.log(`${dog.name): woof woof`)
    const leo = { name: "Leo", owner: "Chris" }
    sayWoof(leo) // TS won't complain about extra "owner" property, but Flow will. 
     ``` 
    is perfectly fine. (TS does actually demand no extra properties when assigning an object literal to a type, a pragmatic, extra-check they call [excess property checking](https://www.typescriptlang.org/docs/handbook/2/objects.html#excess-property-checks). 


IMO Flow's enthusiasm for exact types is paranoid and unhelpful. TS's behavior is more natural, IMO. For reference: https://github.com/microsoft/TypeScript/issues/12936#issuecomment-284590083 

### Resolution
I resolved the issue by bending to Flow's demand :shrug:. There's probably also a way to typecast it to QueryConfig, or suppress the error with a `$FlowFixMe`.

Reply:
Basically all i got from this one was TS>Flow

Reply:
(kidding - makes sense, although it doesn't make sense. If that... makes sense)